.github/setup-node/action.yml

```diff
@@ -28,7 +28,7 @@ runs:
           uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
           with:
               path: '**/node_modules'
-              key: node_modules-${{ runner.os }}-${{ steps.node-version.outputs.NODE_VERSION }}-${{ hashFiles('package-lock.json') }}
+              key: node_modules-${{ runner.os }}-${{ runner.arch }}-${{ steps.node-version.outputs.NODE_VERSION }}-${{ hashFiles('package-lock.json') }}
```

Comment:
What is `runner.arch` and why is it needed now?

Reply:
`runner.arch` is a variable part of the [runner context](https://github.com/github/docs/blob/main/content/actions/learn-github-actions/contexts.md#runner-context) that returns the current architecture of the machine where it is running. For this case `X64`.

For this PR the key it returns: `node_modules-macOS-X64-v20.12.2-ae23e9300c7a80d4e1aa1bfc1a0c6668f8039ccd9867669545e6617f430c4805`

>why is it needed now?

I saw the current image we use was updated [last week](https://github.com/actions/runner-images/releases/tag/macOS-12%2F20240412.2) I also believe the transition to M1+ machines might be related. I think it's best to have the architecture to avoid future issues.